### PR TITLE
Flesh out security context documentation

### PR DESCRIPTION
**Pull request by erictune**

### CODE REVIEW

**Comment by erictune** [comment]
@swagiaal copied from https://github.com/kubernetes/kubernetes/pull/21792

**Comment by erictune** [comment]
@swagiaal I listed you as the author.  Hope that is okay.

**Comment by swagiaal** [comment]
@erictune thank you for doing this! It has been on my list but just kept getting pushed down.

File: docs/user-guide/security-context.md [hunk 1]
```diff
@@ -1,4 +1,82 @@
...
+`seLinuxOptions` are applied to the volume as follows:
+
+#### `fsGroup`
+
+Volumes which support ownership management are modified to be owned
+and writable by the GID specified in `fsGroup`. See the
+[Ownership Management design document](../proposals/volume-ownership-management.md)
+for more details.
+
+#### `seLinuxOptions`
```
Comment (pmorie) [inline docs/user-guide/security-context.md]: should be `selinuxOptions`

Comment (erictune) [inline docs/user-guide/security-context.md]: fixed

**Comment by pmorie** [comment]
This LGTM to me as a start.  We probably want more detail, but this is prefect to go in for now.

File: docs/user-guide/security-context.md [hunk 2]
```diff
@@ -1,4 +1,80 @@
----
----
+A security context defines the operating system security settings (uid, gid, capabilities, SELinux role, etc..) applied to a container. See [security context design](https://github.com/kubernetes/kubernetes/blob/{{page.githubbranch}}/docs/design/security_context.md) for more details.
```
Comment (johndmulhausen) [inline docs/user-guide/security-context.md]: You have to lead the file with these lines:

---
---

That's just one of the rules of GitHub Pages/Jekyll, I'm afraid.

File: docs/user-guide/security-context.md [hunk 3]
```diff
@@ -1,4 +1,80 @@
...
+  # specification of the pod’s containers
+  # ...
+  securityContext:
+    fsGroup: 1234
+    supplementalGroups: [5678]
+    seLinuxOptions:
+      level: "s0:c123,c456"
+```
+
+Please refer to the [API documentation](https://htmlpreview.github.io/?https://github.com/kubernetes/kubernetes/blob/HEAD/docs/api-reference/v1/definitions.html#_v1_podsecuritycontext) for a detailed listing and
```
Comment (johndmulhausen) [inline docs/user-guide/security-context.md]: Change link to: /docs/api-reference/v1/definitions/#_v1_podsecuritycontext

File: docs/user-guide/security-context.md [hunk 4]
```diff
@@ -1,4 +1,80 @@
...
+
+Another functionality of pod level security context is that it applies
+those settings to volumes where applicable. Specifically `fsGroup` and
+`seLinuxOptions` are applied to the volume as follows:
+
+#### `fsGroup`
+
+Volumes which support ownership management are modified to be owned
+and writable by the GID specified in `fsGroup`. See the
+[Ownership Management design document](../proposals/volume-ownership-management.md)
```
Comment (johndmulhausen) [inline docs/user-guide/security-context.md]: Change link to:
https://github.com/kubernetes/kubernetes/blob/{{page.githubbranch}}/docs/proposals/volume-ownership-management.md

File: docs/user-guide/security-context.md [hunk 5]
```diff
@@ -1,4 +1,80 @@
...
+      # The container definition
+      # ...
+      securityContext:
+        privileged: true
+        seLinuxOptions:
+          level: "s0:c123,c456"
+```
+
+Please refer to the
+[API documentation](https://htmlpreview.github.io/?https://github.com/kubernetes/kubernetes/blob/HEAD/docs/api-reference/v1/definitions.html#_v1_securitycontext)
```
Comment (johndmulhausen) [inline docs/user-guide/security-context.md]: Change link to: /docs/api-reference/v1/definitions/#_v1_securitycontext

**Comment by erictune** [comment]
@johndmulhausen addressed all comments.

**Comment by johndmulhausen** [comment]
Thanks, Eric. Much appreciated :)

**Pull request merged**